bot.py

```diff
@@ -843,6 +860,8 @@ def make_commands(self):
         except (InputException, ReleaseException) as ex:
             log.exception("A BotException was raised:")
             await self.say(channel_id=channel_id, text="Oops! {}".format(ex))
+        except ResetException:
+            loop.stop()
```

Comment:
Does this actually stop and restart the process?

Reply:
Yes, it stops the loop which causes a `CancelledError` to be raised, which will interrupt the process. If we just raised an exception here it would just be logged and ignored.

Reply:
It won't restart the process but heroku will do that for processes which exit